### ISSUE TITLE
BatchJobsParam should load BatchJobs configuration

### ISSUE THREAD

**Issue by DarwinAwardWinner** [Bioconductor/BiocParallel#38]
When the BatchJobs package is loaded, it automatically loads the user's custom configuration from the BatchJobs config files. But if you just load BiocParallel and use a BatchJobsParam, this config is never loaded, unless you also explicitly do `library(BatchJobs)` in your own code. We should fix this.

**Comment by vobencha** [comment]
Version 1.1.19 has new registry defaults:
SnowParam, SerialParam for Windows
MulticoreParam, SnowParam and SerialParam for non-windows

BatchJobs is now loaded when a BatchJobsParam is created and the user's config is read.

**Comment by HenrikBengtsson** [comment]
Can this issue be closed?

**Comment by mllg** [comment]
Sorry, I did not meant to close this one. This issue should have been resolved in the linked commit. I'll talk with @berndbischl tomorrow and schedule an update for CRAN.

**Comment by mllg** [comment]
NB: My changes should not break BP. The introduced fix in BatchJobsParam can be removed in a future version.

**Comment by vobencha** [comment]
Hi Michel,
I think I introduced this problem and should have closed the issue with my commit on March 18. In an effort to lighten the NAMESPACE / load time I was playing around with loading packages with bpstart() vs when the constructor was called. One of the side effects was that the config was not read when BatchJobsParam() was created; I reverted that change and returned it to the original behavior.

In [tudo-r/BatchJobs@179a84e](https://github.com/tudor/BatchJobs/commit/179a84eb2415306ce14e1d25723cad31623b4805)  it looks like you've taken this one step further and BatchJobs 1.7 reads the config file when loading the NAMESPACE. Nice. Will you remove the work around in BiocParallel (BatchJobsParam) once BatchJobs 1.7 propagates?

Thanks.
Valerie
